js/test/ValidateSPV.test.js

```diff
@@ -159,4 +171,89 @@ describe('ValidateSPV', () => {
       }
     });
   });
+
+  describe('#validateHeader', () => {
+    it('returns true if all elements of the bitcoin header are valid', () => {
+      for (let i = 0; i < validProof.length; i += 1) {
```

Comment:
`validProof` is an object
`validProof.length` is `undefined`
so your for loop is never running

Reply:
fixed